### PR TITLE
WT-5767 Fix search_near invocations for history store

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
We currently have some code to position our cursor at the end of a key/timestamp range in the history store.

The implementation involves search_near running in a loop. So if we do search_near and land on the next key, we call prev to get back to the latest record for our timestamp. If we're still ahead of our key, that means we raced with another insert and we repeat this process. What should really happen is that we call prev in a loop to get to the right spot since this is much cheaper than doing a full-blown search again.

We also have a few spots where we're using our own search_near invocation without this outer loop to guard against races. We should either generalize the cursor positioning function to work for this use case too or we should add a loop to the current invocation to make it correct.

### CODE REVIEW

**Comment by tetsuo-cpp** [comment]
> lgtm - should __hs_delete_key_int call __wt_hs_cursor_position directly as well?

Yep, I've been waiting to merge my other change. I'll do that now.

**Comment by tetsuo-cpp** [comment]
@agorrod 67f996f

File: src/history/hs.c [hunk 1]
```diff
@@ -1046,36 +1049,47 @@ __hs_delete_key_int(WT_SESSION_IMPL *session, uint32_t btree_id, const WT_ITEM *
...
 
     hs_cursor = session->hs_cursor;
-    hs_cursor->set_key(hs_cursor, btree_id, key, WT_TS_NONE, 0);
-    /*
-     * FIXME: This should be done in a loop in order to deal with the possibility of racing with
-     * other inserts. Perhaps further generalize the cursor positioning function.
-     */
+    WT_RET(__wt_scr_alloc(session, 0, &srch_key));
+
+    hs_cursor->set_key(hs_cursor, btree_id, key, WT_TS_NONE, (uint64_t)0);
```
Comment (luke-pearson) [inline src/history/hs.c]: Why are you casting to (uint64_t)?

Comment (tetsuo-cpp) [reply src/history/hs.c]: I had to do it here: https://github.com/wiredtiger/wiredtiger/pull/5362#discussion_r388557382
So it should also be necessary here.

Comment (tetsuo-cpp) [reply src/history/hs.c]: Also we're doing it in master for lookaside: https://github.com/wiredtiger/wiredtiger/blob/master/src/cache/cache_las.c#L831

**Pull request merged**